runtime/autoload/dist/vimindent.vim

```diff
@@ -1078,6 +1052,24 @@ def ContinuesBelowBracketBlock( # {{{3
 enddef
 
 def IsInside(lnum: number, syntax: string): bool # {{{3
+    if syntax == 'EnumBlock'
+        var cur_pos = getpos('.')
+        cursor(lnum, 1)
+
+        var enum_pos = search('^\C\s*enum\>\s', 'bnW')
+        var endenum_pos = search('^\C\s*endenum\>\s', 'bnW')
```

Comment:
```suggestion
        var enum_pos = search('^\C\s*\%(export\s\)\=\s*enum\s\+\S\+', 'bnW')
        var endenum_pos = search('^\C\s*endenum\>', 'bnW')
```

The flag combination for `search()` can be tricky; the above  
code seems to follow in `PrevCodeLine()`'s footsteps.  If the  
cursor is inside `EnumBlock`, shouldn't it seek _forward_ for  
`endenum` and _backward_ for `enum`?  Will it work when there is  
more than one `enum` definition in the same file?

Reply:
For the `export enum` case I think we should also map 'enum' with 'export' modifier. It is previously uncovered.

```vimscript
--- a/runtime/autoload/dist/vimindent.vim
+++ b/runtime/autoload/dist/vimindent.vim
@@ -172,6 +172,7 @@ const MODIFIERS: dict<string> = {
     def: ['export', 'static'],
     class: ['export', 'abstract', 'export abstract'],
     interface: ['export'],
+    enum: ['export'],
 }
```

Reply:
> If the
cursor is inside EnumBlock, shouldn't it seek forward for
endenum and backward for enum? 

No. This logic may fail. Consider a case

```vim script
enum Color
    Red,
    Green,
    Blue
endenom                  #         <-- Missed

var foo: number = 1 #         <-- This line may fail

enum Subject           #         <-- Missed
    Math,
    Physics
endenum
```
In this case, If we seek for `enum` backward and `endenum` forward from the marked line, we might miss the `endenum` right above and the `enum` right below. As a result, the function could incorrectly conclude that the marked line is inside an EnumBlock.

Reply:
I just noticed that I did make a mistake in the comparison expression.

(I thought the return value when searching backward was relative to the current position… Bruh…)

Thank you again for making me rethink.

Reply:
Fair enough.  It escaped my notice that the cursor is only  
moved once, hence the need to search in one direction.

There is still some room for improvement.

We can do better by avoiding the second lookup as soon as it  
is known that there is no `enum` above.  It is also worth the  
effort to verify that the found `enum` and/or `endenum` line is  
not the current line (remembering about `NonCommentedMatch`'s  
[timeouts](https://github.com/vim/vim/blob/157397edffbc5c375ce54e5e26ee0cabd30a8c73/runtime/autoload/dist/vimindent.vim#L1271)); otherwise, to reject this line for the related  
indentation.  See #16310.